pennylane/pulse/parametrized_evolution.py

```diff
@@ -472,6 +474,53 @@ def hash(self):
             )
         )
 
+    def _flatten(self):
+        """Unpacking the ParametrizedEvolution pytree into a flat tuple"""
+
+        data = self.data
+        metadata = (tuple(self.t), self.H, self.hyperparameters, self.odeint_kwargs)
```

Comment:
How can I make hyperparameters and odeint_kwargs hashable?

Reply:
Since we know what the only two things in hyperparameters are, could we handle them like this?

```suggestion
        metadata = (tuple(self.t), self.H, self.hyperparameters["complementary"], self.hyperparameters["return_intermediate"], self.odeint_kwargs)
```

and then in `_unflatten` we pass `metadata[2]` to `complementary` and `metadata[3]` to `return_intermediate`

Reply:
For `odeint_kwargs` its a little harder since we don't know how many or what they will be called

Reply:
How about something like this:

```
sorted_odeint_kwargs = sorted(op.odeint_kwargs.items())
odeint_kwargs = tuple([kwarg for kwarg, val in sorted_odeint_kwargs] + [val for val, val in sorted_odeint_kwargs])
```
to create a hashable tuple with all the info, and then in `_unflatten`:

```
idx = int(len(odeint_kwargs)/2)
keys = list(odeint_kwargs)[:idx]
vals = list(odeint_kwargs)[idx:]
odeint_kwarg_dict = {k: v for k, v in zip(keys, vals)}
odeint_kwarg_dict
```
and pass that to the operator init?

Reply:
Used a solution Christina suggested on slack that just creates a tuple of tuples from dict.items()